tests/integration/examples/conftest.py

```diff
@@ -21,7 +21,7 @@
 from zenml.stack.stack import Stack
 
 
-@pytest.fixture(scope="module", autouse=True)
+@pytest.fixture(scope="function", autouse=True)
```

Comment:
I don't know if this works as you think it does, because you can't define test requirements at function level in the test harness configuration. The only thing that this change gives you is a clean stack at the level of each function that uses the fixture, but you'll still get the same exact stack configuration for all functions inside a module.

Reply:
I think I understand now that it's exactly what you need: a clean stack for each function in a module, even if it has the same configuration. But why ?

Reply:
Some examples have the option to run with various configurations (actually right now only the huggingface one, for which one of them is disabled) and having this a module scope messed with defining both of those test cases in the same file. Is there any reason not to have this function scoped, seems like the safer solution that works in all scenarios?

Reply:
I can't think of a reason not to have it function scoped.